### PR TITLE
machine: Allow arbitrary matches with journal_matches

### PR DESCRIPTION
**Pull request by mvollmer**
So that we can also look at messages with GLIB_DOMAIN=cockpit-bridge,
say.

### CODE REVIEW

File: machine/machine_core/machine.py [hunk 1]
```diff
@@ -187,7 +187,9 @@ def journal_messages(self, syslog_ids, log_level, cursor=None):
             cursor_arg = ""
 
         cmd = "journalctl 2>&1 %s -o cat -p %d %s || true" % (cursor_arg, log_level, matches)
+        print("J", cmd)
         messages = self.execute(cmd).splitlines()
+        print("M", messages)
```
Comment (martinpitt) [inline machine/machine_core/machine.py:192]: Debugging leftover? If not, this shoudl be a bit more verbose.

**Comment by martinpitt** [comment]
Running two tests against this to make sure it doesn't break.

**Review by martinpitt — CHANGES_REQUESTED** [review]
See above

File: machine/machine_core/machine.py (hunk 1 above)
Comment (mvollmer) [reply machine/machine_core/machine.py:192]: Yes, oops.

Comment (mvollmer) [reply machine/machine_core/machine.py]: Fixed.

**Comment by mvollmer** [comment]
What about changing the meaning of the syslog_ids argument of  `journal_messages`?  We can re-define it to be an array of match expressions like "_PID=1234" and for compatibility translate strings without a "FIELD=" into "SYSLOG_IDENTIFIER=...". I think that would be way cleaner.

**Comment by mvollmer** [comment]
> What about changing the meaning of the syslog_ids argument of `journal_messages`?

I'll do this.

**Comment by martinpitt** [comment]
@mvollmer: That would feel rather hackish IMHO -- isn't that exactly what `match` is for?

**Comment by mvollmer** [comment]
> @mvollmer: That would feel rather hackish IMHO -- isn't that exactly what `match` is for?

Yes, we would remove `match`.  I think it is awkward to have two arguments that do the same thing.

**Comment by martinpitt** [comment]
@mvollmer : But they don't do the same thing -- certainly the new `match` is more powerful, so ISTM that we can port stuff over to `match` and then drop the old `syslog_ids` parameter once nothing uses it any more. I guess there's not actually that many consumers, so maybe you can just do it in one go.

**Comment by mvollmer** [comment]
Tests have run twice.

**Review by martinpitt — APPROVED** [review]
Ack, thanks! Still triggering two integration tests to make sure these work.

**Pull request merged**